src/dbnode/storage/namespace.go

```diff
@@ -501,6 +505,10 @@ func (n *dbNamespace) Tick(c context.Cancellable, tickStart time.Time) error {
 		return nil
 	}
 
+	n.RLock()
+	nsCtx := namespace.Context{Schema: n.schemaDescr}
```

Comment:
In some places you create this with an ID and in some places you don't. Is that intentional? Should we have a constructor function or cache an up-to-date version next to schemaDescr?

Reply:
good catch, I will try to make it consistent (always set a nsID). Though in many places, only schema is ever used.